crates/api_common/src/utils.rs

```diff
@@ -540,15 +540,9 @@ pub fn local_site_rate_limit_to_rate_limit_config(
   })
 }
 
-pub fn local_site_to_slur_regex(local_site: &LocalSite) -> Option<LemmyResult<Regex>> {
-  build_slur_regex(local_site.slur_filter_regex.as_deref())
-}
-
-pub fn local_site_opt_to_slur_regex(local_site: &Option<LocalSite>) -> Option<LemmyResult<Regex>> {
-  local_site
-    .as_ref()
-    .map(local_site_to_slur_regex)
-    .unwrap_or(None)
+pub async fn slur_regex(context: &LemmyContext) -> LemmyResult<Regex> {
+  let local_site = LocalSite::read(&mut context.pool()).await.ok();
+  build_and_check_regex(local_site.and_then(|s| s.slur_filter_regex).as_deref())
```

Comment:
You now could be reading local_site multiple times per endpoint, which was one of the reasons local_site was included as a param in the first place.

Reply:
Reading local site is already cached: https://github.com/LemmyNet/lemmy/blob/main/crates/db_schema/src/impls/local_site.rs#L20

Reply:
Oh nice, didn't realize that.